third_party/nvfuser/csrc/ops/alias.cpp

```diff
@@ -371,4 +372,281 @@ TensorView* transpose(TensorView* x) {
+  Val* right_pad = concat_ext;
+  std::vector<Val*> expanded_inputs(inputs.size());
+  for (const auto input_idx : c10::irange(inputs.size())) {
+    const auto& inp_dom = inp_doms.at(input_idx);
+    std::vector<IterDomain*> root_ids(ndims);
+    std::vector<IterDomain*> rfactor_ids(ndims);
+    std::vector<Val*> pad_widths(ndims * 2);
+    for (const auto dim : c10::irange(ndims)) {
+      auto inp_root_id = inp_dom.at(dim);
+      IterDomain* out_root_id = nullptr;
+      IterDomain* out_rf_id = nullptr;
+      if (dim != cat_dim) {
+        out_root_id = inp_root_id->cloneWithoutRFactor();
+        out_rf_id = out_root_id;
+        pad_widths.at(dim * 2) = FusionGuard::getCurFusion()->zeroVal();
+        pad_widths.at(dim * 2 + 1) = FusionGuard::getCurFusion()->zeroVal();
+      } else {
+        // Resize the root ID so that it has the same extent as the
+        // concatenated ID. The expansion of both left and right sides
+        // is done so that this input tensor is positioned in a way
+        // that corresponds to the concatenated dimension. For
+        // example, the first input should be at the
+        // left-most position, so it is expaned only at the right side
+        // with the expansion factor of
+        // (total_concatenated_domain_extent -
+        // extent_of_the_input_tensor). Similarly, the second tensor
+        // is expanded by extent_of_the_input_tensor at its left side,
+        // and by (total_concatenated_domain_extent -
+        // extent_of_the_input_tensor - extent_of_the_second_tensor).
+        //
+        // TODO: what to do if inp_id is not a normal iterdomain, i.e.,
+        // broadcast, partial, etc? For now, assume it's a normal
+        // IterDomain.
+        TORCH_INTERNAL_ASSERT(
+            inp_root_id->getIterType() == IterType::Iteration &&
+                !inp_root_id->maybePartial(),
+            "Unsupported IterDomain to concatenate: ",
+            inp_root_id->toString());
+        out_root_id =
+            IterDomainBuilder(inp_root_id).is_rfactor_domain(true).build();
+        // The right pad of the last tensor is just zero
+        right_pad = input_idx < inputs.size() - 1
+            ? sub(right_pad, inp_root_id->getMaybeExpandedExtent())
+            : FusionGuard::getCurFusion()->zeroVal();
+        out_rf_id = IterDomain::resize(out_root_id, left_pad, right_pad, true);
+        pad_widths.at(dim * 2) = left_pad;
+        pad_widths.at(dim * 2 + 1) = right_pad;
+        left_pad = add(left_pad, inp_root_id->extent());
+      }
+      root_ids.at(dim) = out_root_id;
+      rfactor_ids.at(dim) = out_rf_id;
+    }
+
+    auto resized_inp = IrBuilder::create<TensorView>(
+        IrBuilder::create<TensorDomain>(
+            root_ids,
+            rfactor_ids,
+            rfactor_ids,
+            TensorDomain::getContiguousContiguity(rfactor_ids)),
+        dtype);
+
+    IrBuilder::create<PadOp>(resized_inp, inputs.at(input_idx), pad_widths);
```

Comment:
Why not use the `pad` function above?

Reply:
Thanks for the suggestion. Done: 8ad222c0fc76c48c885bfdfba14da5d96e00ab34